Index out of range when calling log methods with logger's name "xybor.xyplatform.main"

Version: v1.0.0
```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/xybor/xyplatform/xylog.(*Logger).callHandlers(0xc0000841b0, {{0xc000026030, 0x23}, 0x62fdb1c8, {0x4e55f9, 0x7}, {0xc000018121, 0x4}, {0x4b142a, 0x5}, ...})
        /home/toto/go/pkg/mod/github.com/xybor/xyplatform@v1.0.0/xylog/logger.go:166 +0x154
github.com/xybor/xyplatform/xylog.(*Logger).handle(0xc0000841b0, {{0xc000026030, 0x23}, 0x62fdb1c8, {0x4e55f9, 0x7}, {0xc000018121, 0x4}, {0x4b142a, 0x5}, ...})
        /home/toto/go/pkg/mod/github.com/xybor/xyplatform@v1.0.0/xylog/logger.go:152 +0x87
github.com/xybor/xyplatform/xylog.(*Logger).log(0xc0000841b0, 0x8?, {0xc000018110, 0xb})
        /home/toto/go/pkg/mod/github.com/xybor/xyplatform@v1.0.0/xylog/logger.go:146 +0x105
github.com/xybor/xyplatform/xylog.(*Logger).Debug(0xc000070f70?, {0x4b1b17, 0x8}, {0xc000070f60, 0x1, 0x1})
        /home/toto/go/pkg/mod/github.com/xybor/xyplatform@v1.0.0/xylog/logger.go:82 +0x85
main.main()
        /home/toto/Golang/main/main.go:12 +0x6a
exit status 2
```

--- a/xylog/logger.go
+++ b/xylog/logger.go
@@ -192,7 +192,7 @@ func (lg *Logger) callHandlers(record LogRecord) {
 	var found = 0
 	for c != nil {
 		for i := range c.handlers {
-			lg.handlers[i].handle(record)
+			c.handlers[i].handle(record)
 			found += 1
 		}
 		c = c.parent

--- a/xylog/logger_test.go
+++ b/xylog/logger_test.go
@@ -116,6 +116,20 @@ func TestLoggerLogMethods(t *testing.T) {
 	}
 }
 
+func TestLoggerCallHandlerHierarchy(t *testing.T) {
+	var expectedMessage = "foo"
+	var handler = xylog.NewHandler("", &CapturedEmitter{})
+	var logger = xylog.GetLogger(t.Name())
+	logger.SetLevel(xylog.DEBUG)
+	logger.AddHandler(handler)
+
+	logger = xylog.GetLogger(t.Name() + ".main")
+	capturedOutput = ""
+	logger.Info(expectedMessage)
+	xycond.MustEqual(capturedOutput, expectedMessage).
+		Testf(t, "%s != %s", capturedOutput, expectedMessage)
+}
+
 func TestLoggerLogNoHandler(t *testing.T) {
 	var logger = xylog.GetLogger(t.Name())
 	logger.SetLevel(xylog.DEBUG)

